### DISCUSSION TITLE
Resource Server to Resource Server REST API Call

### DISCUSSION THREAD

**Discussion by Aleson95** [keycloak/keycloak#14816] [Q&A] [answered]
In our microservices architecture there are 2 types of services / applications / clients that are registered to Keycloak:
-public client, this is our frontend applications using vue js
-bearer-only, this is our backend applications (resource servers) using java spring boot
I have a usecase where I need my backend application A calling backend application B or vice versa to be accepted or respected but I do not want to use the access token from the public client to be pass to the other microservice because each resource server has their own unique set of roles that were used by their respective microservices to authorized each API endpoint.
I just want to make sure that the REST API call is coming from the client that registered in keycloak which means this microservices is part of the services that I try to build.
Here is a diagram about the use case I talk about
![microservices](https://user-images.githubusercontent.com/42630865/194826165-ffd4cb78-dd79-4d29-9a26-789e6a889699.jpg)
Is there anything like client authentication, where I can authenticate using client id and secret and get an access token to be used to access any or restricted API endpoint of the specific client id?

**Comment by jbman** [comment, marked as answer]
You can use a confidential client with "Service Accounts Enabled". Such a client acts as a technical user with an own set of roles.
Such a client uses OAuth 2.0 Client credentials grant to get an access token.

**Reply by Aleson95** [reply]
hello thank you for the reply this is what I needed thank you
